### PR TITLE
Container build artifacts.

### PR DESCRIPTION
**Pull request by supriya-premkumar**
Changes:
            1. Adds docker support for building kv and client.
            2. Uses multistage builds to copy over built artifacts from the builder.
            3. Adds go mod tidy to clean up unused deps.
            4. Lints and cleans up raft.proto to ensure offsets are not skipped.
            5. Deployable images live at supriyapremkumar:v0.1
            6. Enables docker builds in CircleCI

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -7,7 +7,7 @@ require (
 	github.com/boltdb/bolt v1.3.1
 	github.com/fatih/color v1.9.0
 	github.com/gogo/protobuf v1.3.1
-	github.com/golang/protobuf v1.3.3
+	github.com/golang/protobuf v1.4.1
```
Comment (renga86) [inline go.mod:10]: qq: I recollect Chen saying the latest version breaks some functionality. Did you get a chance to test it out with the built docker image ?

Comment (alex8937) [reply go.mod:10]: Yeah. I remember protobuf 1.4 is not compatiable with gogo 1.3.1

File: raftpb/raft.pb.go [hunk 2]
```diff
@@ -1,542 +1,801 @@
 // Code generated by protoc-gen-go. DO NOT EDIT.
+// versions:
+// 	protoc-gen-go v1.24.0-devel
+// 	protoc        v3.11.2
```
Comment (alex8937) [inline raftpb/raft.pb.go:4]: Again, DO NOT edit *pb.go file as shown above. It will be anyway rewritten when rebuild.

Comment (supriya-premkumar) [reply raftpb/raft.pb.go:4]: This is *NOT* a manual edit. Its the output generated from protoc builds. The newer version adds compile metadata to the generated files. It is useful for figuring out how the pb.go was generated.

File: go.mod (hunk 1 above)
Comment (supriya-premkumar) [reply go.mod:10]: The compatibility issues are in the enum and oneofs that are used. The reason for this non-compat is due to the fact the go doesn't support unions. 
In our case we are neither using enums or any of the rpcs. So this should be ok. Ran all the tests they look ok.

Let me know if there is anything specific that you're worried about

Comment (varks) [reply go.mod:10]: I used the newer version on kv before I moved back to 1.3.3 and it seems to be fine. As long as you have tested its fine.

**Comment by supriya-premkumar** [comment]
## Results from the container run

Screen Shot 2020-05-31 at 7 02 05 PM

File: go.mod (hunk 1 above)
Comment (supriya-premkumar) [reply go.mod:10]: Yes this works.

**Comment by alex8937** [comment]
Consider modifying README.md as well.

**Pull request merged**